pydantic_ai/models/__init__.py

```diff
@@ -44,14 +44,16 @@ class Model(ABC):
     """Abstract class for a model."""
 
     @abstractmethod
-    def agent_model(
+    async def agent_model(
         self,
         retrievers: Mapping[str, AbstractToolDefinition],
         allow_text_result: bool,
         result_tools: Sequence[AbstractToolDefinition] | None,
     ) -> AgentModel:
         """Create an agent model.
 
+        This is async in case slow/async config checks need to be performed that can't be done in `__init__`.
+
```

Comment:
I'm not sure this needs to be clarified in the docstring, I think it's okay if that's just how it is.